### PR TITLE
RMHDR-263 New Sleep Measure Summaries and Timescales

### PR DESCRIPTION
**Pull request by pranavanba**
## Major Changes

1. Create a function (calcMetrics) to calculate chosen metrics/summary statistics for chosen variables
2. Add scripts to manually calculate:
    1. Standard deviation of Mid-Sleep and Duration variables
    2. Percent of sleep time spent in sleep stages
    3. Percent of sleep StartTime that occurs at typical nighttime (between 8 pm and 2 am)

## Minor Changes

1. New directory structure in `scripts/` dir
    - new `functions/` subdir
    - Moved `fetch-data` script to new `etl/` subdir

### CODE REVIEW

**Comment by pranavanba** [comment]
TODO: sleep regularity index (SRI) and 3 weeks sliding window timescale

File: scripts/sleepSD.R [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  sleeplogs_df %>% 
+  left_join((participants %>% select(ParticipantIdentifier, InfectionFirstReportedDate)), 
+            by = "ParticipantIdentifier")
+
+# Weekly statistics
+weekly_stats <- 
+  list(
+    midsleep = 
+      merged_data %>%
+      group_by(ParticipantIdentifier, WeekStart = floor_date(Date, "week")) %>%
```
Comment (sieberts) [inline scripts/sleepSD.R:55]: Here you seem to be generating weekly stats, but only using weeks starting 3 months post-infection. You don't need to do that. You should only filter by 3-months post-infection for the timescale that uses all data starting 3-months post-infection.

File: scripts/sleepSD.R [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      ungroup()
+  )
+
+# All-time statistics
+alltime_stats <- 
+  list(
+    midsleep =
+      merged_data %>%
+      group_by(ParticipantIdentifier) %>%
+      filter(Date >= (InfectionFirstReportedDate + months(6))) %>%
```
Comment (sieberts) [inline scripts/sleepSD.R:80]: This says alltime, but you're filtering by > 6-months post-infection. There should be 5 different timescales:

1 week (all weeks)
3 weeks  (sliding window--all weeks)
All-time
All-time Starting 3-months post-infection
All-time Starting 6-months post-infection

File: scripts/timeInSleepStages.R [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  summarise(
+    across(.cols = all_of(c("PercentDeep", "PercentLight", "PercentRem")), 
+           .fns = 
+             list(
+               Mean = ~mean(.x, na.rm = TRUE),
+               Median = ~median(.x, na.rm = TRUE),
+               Variance = ~var(.x, na.rm = TRUE),
+               Percentile5 = ~quantile(.x, 0.05, na.rm = TRUE),
+               Percentile95 = ~quantile(.x, 0.95, na.rm = TRUE),
+               Count = ~n()
```
Comment (sieberts) [inline scripts/timeInSleepStages.R:49]: Need to remove NAs.

**Review by sieberts — CHANGES_REQUESTED** [review]

File: scripts/functions/calcMetrics.R [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  summary_functions <-
+    list(
+      mean = ~mean(.x, na.rm = TRUE),
+      median = ~median(.x, na.rm = TRUE),
+      variance = ~var(.x, na.rm = TRUE),
+      percentile5 = ~quantile(.x, 0.05, na.rm = TRUE),
+      percentile95 = ~quantile(.x, 0.95, na.rm = TRUE),
+      count = ~n(),
```
[the remark this answers is not in the document]
Comment (pranavanba) [reply scripts/functions/calcMetrics.R]: Resolved in [0327bc1](https://github.com/Sage-Bionetworks/recover-sleep-measures/pull/2/commits/0327bc15bd751829595d8d56c5f1a611c9538869)

File: scripts/timeInSleepStages.R (hunk 3 above)
Comment (pranavanba) [reply scripts/timeInSleepStages.R]: Resolved in [0327bc1](https://github.com/Sage-Bionetworks/recover-sleep-measures/pull/2/commits/0327bc15bd751829595d8d56c5f1a611c9538869)

File: scripts/sleepSD.R (hunk 2 above)
Comment (pranavanba) [reply scripts/sleepSD.R]: Resolved in [55dc440](https://github.com/Sage-Bionetworks/recover-sleep-measures/pull/2/commits/55dc440dd3683d9c6459679edc5d839f6f04b61f)

File: scripts/sleepSD.R (hunk 1 above)
Comment (pranavanba) [reply scripts/sleepSD.R]: Resolved in [55dc440](https://github.com/Sage-Bionetworks/recover-sleep-measures/pull/2/commits/55dc440dd3683d9c6459679edc5d839f6f04b61f)

File: scripts/sleepSD.R [hunk 5]
```diff
@@ -0,0 +1,96 @@
+source("scripts/fetch-data.R")
+
+participants <- 
+  arrow::open_dataset(
+    s3$path(stringr::str_subset(dataset_paths, "enrolledparticipants$"))
+  ) %>% 
+  select(c("ParticipantIdentifier",
```
[the remark this answers is not in the document]
Comment (pranavanba) [reply scripts/sleepSD.R]: Resolved in [c985b69](https://github.com/Sage-Bionetworks/recover-sleep-measures/pull/2/commits/c985b697a8e1ce8cb729fbb5f8cdadfa8e379b25)

**Pull request merged**